iothub_client/inc/iothub_client_core_common.h

```diff
@@ -234,6 +243,73 @@ extern "C"
+    * @param[in]    userContextCallback  User context pointer set in initial call to IoTHubDeviceClient_LL_SubscribeToCommands.
+    *
+    * @remarks
+    *            The application behavior is undefined if the user calls
+    *            the IoTHubDeviceClient_LL_Destroy function from within any callback.
+    *
+    * @return    IOTHUB_CLIENT_OK upon success or an error code upon failure.
+    */
+    typedef int(*IOTHUB_CLIENT_COMMAND_CALLBACK_ASYNC)(
+                    const char* componentName,
+                    const char* commandName,
+                    const unsigned char* payload,              
+                    size_t size,
+                    const char* payloadContentType,
+                    unsigned char** response,
+                    size_t* responseSize,
+                    void* userContextCallback);
+
+    /**
+    * @brief    Function callback application implements to receive notifications when IoT Hub acknowledges telemetry.
+    *
+    * @param    result                Result of operation sending to IoT Hub.
+    * @param    userContextCallback   Optional user specified context set in call that initiated the message send (for example, 
+    *                                 IoTHubDeviceClient_LL_SendTelemetryAsync.
+    */
+    typedef void(*IOTHUB_CLIENT_TELEMETRY_CALLBACK)(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void* userContextCallback);
+    
+    /**
+    * @brief    Function callback application implements to receive acknowledgements of properties sent from the device to IoT Hub.  
+    *
+    * @param[in]    statusCode            Status code IoT Hub returns when receiving a property or writable property response from the device.  
+    *                                     This corresponds to HTTP status codes.
+    * @param[in]    userContextCallback   Optional user specified context set in the call that sent the properties (for example,
+    *                                     IoTHubDeviceClient_LL_SendPropertiesAsync).
+    * 
+    */
+    typedef void(*IOTHUB_CLIENT_PROPERTY_ACKNOWLEDGED_CALLBACK)(int statusCode, void* userContextCallback);
+
+    /**
+    * @brief    Function callback application implements to process properties received from IoT Hub.
+    *
+    * @param[in]   payloadType          Whether the payload contains all properties from IoT Hub or only the ones that have just been updated.
+    * @param[in]   payload              Raw payload of the request.  This is NOT guaranteed to be a \0 terminated string.
+    * @param[in]   payloadLength        Number of bytes of payload.
+    * @param[in]   userContextCallback  User context pointer set in initial call to retrieving the properties (for example, 
+    *                                   IoTHubDeviceClient_LL_GetPropertiesAsync or IoTHubDeviceClient_LL_GetPropertiesAndSubscribeToUpdatesAsync).
+    */
+    typedef int(*IOTHUB_CLIENT_PROPERTIES_RECEIVED_CALLBACK)(
```

Comment:
This should be a `void` instead of `int` because we don't do anything with return value.  Also brings inline with the underlying `IOTHUB_CLIENT_DEVICE_TWIN_CALLBACK` behavior.